sdk/template/ci.yml

```diff
@@ -34,6 +34,7 @@ stages:
     parameters:
       ServiceDirectory: template
       CtestRegex: azure-template
+      BranchCoverageTarget: 33
```

Comment:
LineCoverageTarget ??  default?

Reply:
Should template project use both LineCoverageTarget and  BranchCoverageTarget with default values?

Reply:
The template project carries some interesting issues with it. 

We expect new projects to use the template project going forward and we expect that the `ci.yml` file will be copy-pasted, likely without much consideration.

If we provide default values in this file those values will likely be copy-pasted without consideration. It is preferable to defer to the default specified in `archetype-sdk-client.yml` so that we can move the coverage floor up across projects without having to modify numerous `ci.yml` files. 

At the same time, branch coverage for template is lower than default baseline so in order to get template to pass, I had to set a lower coverage floor. 

Issue to track improving coverage for template: https://github.com/Azure/azure-sdk-for-cpp/issues/3022